specs/AppLifecycle/StateNotifications/AppLifecycle StateNotifications.md

```diff
@@ -0,0 +1,389 @@
+    {
+        return FALSE;
+    }
+    MSG msg;
+    while (GetMessage(&msg, nullptr, 0, 0))
+    {
+        TranslateMessage(&msg);
+        DispatchMessage(&msg);
+    }
+    return (int)msg.wParam;
+}
+```
+
+In the app-defined callback, the app can do whatever it likes. Typically, it would fetch the current
+status of the particular feature (that is, the battery status, power supply status, energy saver
+status, and so on), and use that information to form its decision. For example, if battery is at a
+low power state and discharging, the app might choose to defer any non-critical background work, in
+the interests of being a good power/battery citizen.
+
+```c++
+void OnBatteryStateChange()
+{
+    BatteryStatus batteryState = PowerManager::BatteryStatus();
+    int remainingCharge = PowerManager::RemainingChargePercent();
+
+    if (batteryState == BatteryStatus::Discharging && remainingCharge < 25)
+    {
+        // We're in a bad battery state, we should pause any non-critical work.
+        PauseNonCriticalWork();
+    }
+    else if (batteryState == BatteryStatus::Charging && remainingCharge > 75)
+    {
+        // Battery is in great shape, let's kick of some high-power work.
+        StartPowerIntensiveWork();
+    }
+}
+```
+
+### App logic based on multiple status values
+
+The individual notifications are relatively low-level, and in some scenarios each notification in
+itself might not provide enough information for the app to make sensible decisions. More likely, the
+app will use multiple PowerManager properties and methods to factor into its decision. For example,
+when it receives a power supply state change notification, it might also want to check the battery
+status before proceeding.
+
+```c++
+void OnPowerStateChange()
+{
+    PowerSupplyStatus powerState = PowerManager::PowerSupplyStatus();
+    BatteryStatus batteryState = PowerManager::BatteryStatus();
+    int remainingCharge = PowerManager::RemainingChargePercent();
+
+    if (batteryState == BatteryStatus::Discharging
+        && remainingCharge < 25
+        && powerState != PowerSupplyStatus::Adequate)
+    {
+        // We're in a bad power/battery state: let's pause any non-critical work.
+        PauseNonCriticalWork();
+    }
+    else if (powerState == PowerSupplyStatus::Adequate ||
+        (batteryState == BatteryStatus::Charging && remainingCharge > 75))
+    {
+        // Power/battery is in great shape, let's kick of some high-power work.
+        StartPowerIntensiveWork();
+    }
+}
+```
+
+### New API combines multiple old APIs
+
+The additional enums, properties and events brought forward for the PowerSettingRegisterNotification
+functionality should be completely seamless and indistinguishable from the existing PowerManager
+members, such that the app can consume these in exactly the same way.
+
+```c++
+PowerManager::ConsoleStatusChanged([](auto&&...)
+{
+    DisplayStatus displayState = PowerManager::DisplayStatus();
```

Comment:
RECOMMEND: s/displayState/status/

Reply:
DISCUSSION: "State" is how things are when you check them.  "Status" is the result of an operation of some kind.

RECOMMEND: Normalize on `-State` since these are properties, not operations

Reply:
Fixed.